qutebrowser/config/config.py

```diff
@@ -187,11 +188,21 @@ def get_reverse_bindings_for(self, mode: str) -> '_ReverseBindings':
     def get_command(self,
                     key: keyutils.KeySequence,
                     mode: str,
-                    default: bool = False) -> str:
+                    default: bool = False,
+                    pattern: urlmatch.UrlPattern = None) -> str:
         """Get the command for a given key (or None)."""
         self._validate(key, mode)
         if default:
+            if pattern is not None:
+                raise cmdutils.CommandError(
+                    "Cannot get defaults with a url pattern!")
             bindings = dict(val.bindings.default[mode])
+        elif pattern is not None:
+            # FIXME is there a way to avoid the to_py here?
+            opt = self._config.get_opt('bindings.commands')
+            bindings = self._config.get_obj_for_pattern(
+                'bindings.commands', pattern=pattern)
+            bindings = opt.typ.to_py(bindings).get(mode)
```

Comment:
This feels like a hack, but I'm not sure if there's a function to get an *internal* config object by pattern (`get_obj_for_pattern` returns the yaml representation).